.github/workflows/check-changelog.yml

```diff
@@ -0,0 +1,45 @@
+name: Check Changelog
+on:
+  pull_request:
+jobs:
+  check-changelog:
+    runs-on: [ubuntu-latest]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3.3.0
+
+      - name: Find Changed Files in changelog.d
+        id: changed-files
+        uses: tj-actions/changed-files@v35
+        with:
+          files: '**/changelog.d/**'
+
+      - name: Enforce New File or DISABLE_CHANGELOG_CHECK
+        uses: actions/github-script@v6
+        with:
+          script: | 
+            function shouldCheckChangelog() {
+              const prDescription = '${{ github.event.pull_request.body }}';
+              return !prDescription.includes("DISABLE_CHANGELOG_CHECK");
```

Comment:
Maybe we should use a label for this? Also makes it easier to see elsewhere.

Reply:
I'm not sure what labels are in this context

Reply:
Or so we'd look for a label named "Requires Changelog"? or "Does not Require Changelog"?

Reply:
I think we want to label the exceptional case, so we'd have a "No Changelog Required" label and the workflow would check for that when deciding whether to complain. We would also need to rerun the workflow if the labels change.